### PR TITLE
Trampoline has 360 rotation

### PR DESCRIPTION
**Pull request by CatWiz**
## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

Attempt 2.
When held, trampoline freely rotates like drill. 
Holding lmb makes it rotate in 45 degree steps, and holding rmb prevents it from rotating completely (allows players to throw tramps at any angles in different directions and place horizontal tramps more easily)

### CODE REVIEW

File: Entities/Structures/Trampoline/TrampolineLogic.as [hunk 1]
```diff
@@ -41,12 +41,29 @@ void onTick(CBlob@ this)
...
 	f32 angle = ray.Angle();
-	angle = angle > 135 || angle < 45? (holder.isFacingLeft()? 135 : 45) : 90;
-	angle -= 90;
+	CControls@ controls = holder.getControls();
 
-	this.setAngleDegrees(-angle);
+	if (point.isKeyPressed(key_action1))
+	{
+		// rotate in 45 degree steps
+		angle = Maths::Floor((angle + 22.5f) / 45) * 45;
```
Comment (eps0003) [inline Entities/Structures/Trampoline/TrampolineLogic.as:51]: The face of the trampoline isn't facing the cursor.
```suggestion
		angle = Maths::Floor((angle - 67.5f) / 45) * 45;
```

File: Entities/Structures/Trampoline/TrampolineLogic.as [hunk 2]
```diff
@@ -41,12 +41,29 @@ void onTick(CBlob@ this)
...
+		this.setAngleDegrees(-angle);
+	}
+	else if (point.isKeyPressed(key_action2))
+	{
+		// set angle to what was on previous tick
+		angle = this.get_f32("old angle");
+		this.setAngleDegrees(angle);
+	}
```
Comment (eps0003) [inline Entities/Structures/Trampoline/TrampolineLogic.as:59]: This should be prioritised over left-clicking, so please move this if statement above the previous if statement.

**Review by eps0003 — CHANGES_REQUESTED** [review]

File: Entities/Structures/Trampoline/TrampolineLogic.as (hunk 2 above)
Comment (CatWiz) [reply Entities/Structures/Trampoline/TrampolineLogic.as]: done

**Review by eps0003 — APPROVED** [review]

**Review by AsuMagic — APPROVED** [review]

**Comment by AsuMagic** [comment]
Superseded by #1359

**Pull request closed without merging**